src/components/settings/settings-custom-labels/settings-custom-labels.js

```diff
@@ -0,0 +1,203 @@
+    const removingDisplayLabels = pickBy(initialDisplayLabels, (value, key) => (formDisplayLabels[key] === undefined));
+
+    return Object.values(removingDisplayLabels).join(', ');
+  }
+
+  onSubmit = (formValues, { getState }) => {
+    const { updateCustomLabels } = this.props;
+    const { initialValues } = getState();
+
+    const removingLabels = this.getRemovingCustomLabels(formValues, initialValues);
+
+    if (removingLabels.length !== 0) {
+      this.setState({
+        modalIsOpen: true,
+        removingLabels,
+      });
+    } else {
+      const formattedFormValues = formatCustomLabelsValues(formValues);
+
+      updateCustomLabels(formattedFormValues);
+    }
+  }
+
+  getToastLabels() {
+    const {
+      customLabels: {
+        errors,
+        isUpdated,
+      },
+    } = this.props;
+
+    if (isUpdated) {
+      return [{
+        message: <FormattedMessage id='ui-eholdings.settings.customLabels.toast' />,
+        type: 'success',
+        id: `update-custom-labels-success-${Date.now()}`
+      }];
+    } else {
+      return errors.map((error) => ({
+        message: error.title,
+        type: 'error',
+      }));
+    }
+  }
+
+  closeModal = () => this.setState({ modalIsOpen: false });
+
+  render() {
+    const {
+      confirmUpdate,
+      customLabels,
+    } = this.props;
+    const {
+      modalIsOpen,
+      removingLabels,
+    } = this.state;
+
+    const customLabelsData = get(customLabels, ['items', 'data'], []);
+    const allCustomLabels = getAllCustomLabelsFields(customLabelsData);
+    const initialValues = this.prepareInitialValues(allCustomLabels);
+
+    return (
+      <Form
+        onSubmit={this.onSubmit}
+        initialValues={initialValues}
+        mutators={arrayMutators}
+        render={(formState) => (
+          <SettingsForm
+            data-test-eholdings-settings-custom-labels
+            id="custom-labels-form"
+            formState={formState}
+            title={<FormattedMessage id="ui-eholdings.resource.customLabels" />}
+            toasts={this.getToastLabels(formState)}
+            toastAction={confirmUpdate}
+          >
+            <Row>
+              <Col xs={4}>
+                <KeyValue label={<FormattedMessage id="ui-eholdings.settings.customLabels.displayLabel" />} />
+              </Col>
+              <Col xs={4}>
+                <KeyValue label={<FormattedMessage id="ui-eholdings.settings.customLabels.publicationFinder" />} />
+              </Col>
+              <Col xs={4}>
+                <KeyValue label={<FormattedMessage id="ui-eholdings.settings.customLabels.textFinder" />} />
+              </Col>
+            </Row>
+
+            <CustomLabelsFields customLabels={allCustomLabels} />
+
+            <NavigationModal
+              label={<FormattedMessage id="ui-eholdings.navModal.areYouSure" />}
+              message={<FormattedMessage id="ui-eholdings.navModal.unsavedChanges" />}
+              when={!isEqual(formState.values, initialValues)}
```

Comment:
I guess the render function is provided with `pristine`  filed. Please, check if you can use it